src/main/java/org/opentripplanner/middleware/OtpMiddlewareMain.java

```diff
@@ -57,58 +54,35 @@ public class OtpMiddlewareMain {
     public static boolean inTestEnvironment = false;
 
     public static void main(String[] args) throws IOException {
-        // Load configuration.
-        ConfigUtils.loadConfig(args);
-
-        // Initialize template engine
-        TemplateUtils.initialize();
+        CommandLineProcessor processor = new CommandLineProcessor();
+        processor.parseArguments(args);
 
+        // Load configuration.
+        ConfigUtils.loadConfig(processor.getConfigFile());
         // Connect to MongoDB.
         Persistence.initialize();
 
-        MonitoredComponent.initializeMonitoredComponentsFromConfig();
-
-        initializeHttpEndpoints();
-
-        // If Middleware started from test class, skip recurring jobs (that can be log heavy).
         if (!inTestEnvironment) {
-            // Schedule Bugsnag jobs to start retrieving Bugsnag event/error data
-            BugsnagJobs.initialize();
-            // Initialize Bugsnag in order to report application errors
+            // Initialize Bugsnag in order to report application errors.
             BugsnagReporter.initializeBugsnagErrorReporting();
-            // Schedule trip history uploads.
-            ConnectedDataManager.scheduleTripHistoryUploadJob();
-
-            // Schedule recurring jobs.
-            // TODO: Determine whether this should go in some other process.
-
-            MonitorAllTripsJob monitorAllTripsJob = new MonitorAllTripsJob();
-            Scheduler.scheduleJob(
-                monitorAllTripsJob,
-                0,
-                1,
-                TimeUnit.MINUTES
-            );
-
-            TripSurveySenderJob tripSurveySenderJob = new TripSurveySenderJob();
-            Scheduler.scheduleJob(
-                tripSurveySenderJob,
-                0,
-                30,
-                TimeUnit.MINUTES
-            );
-
-            if (TripSurveyUploadJob.isConfigured()) {
-                LOG.info("Scheduling trip survey upload every day");
-                TripSurveyUploadJob tripSurveyUploadJob = new TripSurveyUploadJob();
-                Scheduler.scheduleJob(
-                    tripSurveyUploadJob,
-                    0,
-                    1,
-                    TimeUnit.DAYS
-                );
+            Set<RecurringJob> recurringJobs = processor.getRecurringJobs();
+            if (!recurringJobs.isEmpty()) {
+                recurringJobs.forEach(RecurringJob::schedule);
+                try {
+                    // Block the main thread until the scheduler is terminated to prevent the application from
+                    // immediately stopping.
+                    Scheduler.schedulerService.awaitTermination(Long.MAX_VALUE, TimeUnit.MILLISECONDS);
+                } catch (InterruptedException e) {
+                    LOG.info("Shutting down.", e);
+                    Thread.currentThread().interrupt();
+                }
             }
         }
+
+        if (processor.hasEndPoints()) {
```

Comment:
Move this block to before the scheduler stuff. Otherwise, because of the thread blocking above, this block is never executed (until termination), and therefore no endpoints are available.